src/app/shared-services/timetable/timetable.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core'
+import { Inference } from 'src/app/model/inference'
+import { TimetableEntry } from 'src/app/model/timetable'
+import { AbstractBackgroundService } from '../background/AbstractBackgroundService'
+import { BackgroundService } from '../background/background.service'
+import { SqliteService } from '../db/sqlite.service'
+import { NotificationService } from '../notification/notification.service'
+import { NotificationType } from '../notification/types'
+import { TrajectoryService } from '../trajectory/trajectory.service'
+import { Timetable } from './timetable'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TimetableService extends AbstractBackgroundService {
+  protected foregroundInterval = 720
+  protected backgroundInterval = 15
```

Comment:
There are varying intervals in our background-services (`InferenceService` vs `TimetableService`) – is this a good idea, given that we only have one instance of background-fetch as stated here: https://github.com/sitcomlab/simport-learning-app/issues/101#issuecomment-986581065

Reply:
I'm not that sure about that as both intervals are still used in the corresponding `triggerBackgroundFunctionAsTask` and `triggerBackgroundFunctionAsFetch` functions to check whether the execution is in the correct schedule.